packages/plugins/slugManager/manager.ts

```diff
@@ -85,25 +83,33 @@ export function createManager(config: InternalSlugManagerConfig) {
   }
 
   async function webhook(event: WebhookEvent) {
-    if (!(isEntityCreate(event) || isLangProfileCreate(event))) return;
+    if (!(isEntityCreate(event) || isFieldUpdate(event, slugGeneratorFnFields)))
+      return;
 
     if (entityTypes.length) {
       if (!entityTypes.includes(event.primaryProfile.meta.entityType)) return;
     }
 
     if (searchIds.length) {
-      if (!(await api.savedSearchesIncludeEntity(searchIds, event.entityId))) {
-        return null;
-      }
+      if (
+        !(event.primaryProfile.savedFilters || []).some((filter) =>
+          searchIds.includes(filter)
+        )
+      )
+        return;
     }
 
     const lang = event.changedFields.language;
-    const slug = slugGeneratorFn
-      ? slugGeneratorFn(lang, event.primaryProfile)
-      : lang in slugFormatLocaleOverrides
-      ? slugFormatLocaleOverrides[lang]
-      : slugFormat;
-
+    const profile =
+      lang == event.primaryProfile.meta.language
+        ? event.primaryProfile
+        : event.languageProfiles.find(
+            (profile) => profile.meta.language == lang
+          );
+    const slug =
+      typeof slugFormat == "string"
+        ? slugFormat
+        : slugFormat.generatorFn(lang, profile!);
```

Comment:
instead of `profile!` I think we could throw an error here, that way hopefully whatever shows up in the platform logs is more explicit about what went wrong